app/src/main/java/org/mozilla/fenix/home/recentsyncedtabs/RecentSyncedTabFeature.kt

```diff
@@ -82,10 +84,14 @@ class RecentSyncedTabFeature(
 
     // UI will either not be displayed if not authenticated (DefaultPresenter.start),
     // or the display state will be tied directly to the success and error cases.
-    override fun stopLoading() = Unit
+    override fun stopLoading() {
+        isSyncing = false
+    }
 
     override fun onError(error: SyncedTabsView.ErrorType) {
-        store.dispatch(AppAction.RecentSyncedTabStateChange(RecentSyncedTabState.None))
+        if ((!isSyncing && error == SyncedTabsView.ErrorType.NO_TABS_AVAILABLE) || error.isFatal()) {
```

Comment:
Instead of having an `isSyncing` variable, could we check `store.state.recentSyncedTabState` to see if it's loading? Not saying this is necessarily the correct thing to do, but in prior reviews I had an inkling that maybe `recentSyncedTabState` should be tri-state with `stopLoading` implemented.

Reply:
I'll update it to inspect the store's state, but would you mind clarifying about the tri-state? Maybe it's time to revisit that since we've already talked about it at least once. 

Is the idea to introduce an error state? From the consumer's perspective, there isn't (currently) a desire to display error messages, which is why I've been defaulting to `NONE` on error.

Reply:
It might be something more like `START`( or `NONE`), `LOADING`, `DONE`. I don't know if this will necessarily be correct, but instead of having `NONE`, we need to have a state that differentiates between whether or not loading was attempted.

Reply:
There currently are three states that I think map pretty well to those: `NONE`, `LOADING`, and `SUCCESS`. Switching `SUCCESS` to a `DONE` state that had either successes or failures could potentially be an improvement, but I'm not sure what it benefits over the current implementation.

Honestly I think the underlying [Controller](https://searchfox.org/mozilla-mobile/source/android-components/components/feature/syncedtabs/src/main/java/mozilla/components/feature/syncedtabs/controller/DefaultController.kt) implementation's usage of `stopLoading` is generally a bit confusing. The only case it's ever really useful IMO is the one where an account isn't authenticated, but `refreshSyncedTabs` shouldn't be called unless an account is authenticated anyway.